### PR TITLE
Weekly Patch Release v.1.2.7 [full merge, no squash]

### PR DESCRIPTION
**Pull request by kaushikb11**
## PR review

related to #6083

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
lgtm, just fix tests...

**Review by ananthsub — APPROVED** [review]

**Review by Borda — CHANGES_REQUESTED** [review]
just till tests are fixed 🐰

**Comment by kaushikb11** [comment]
@Borda There you go! 🎉

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 1]
```diff
@@ -203,12 +203,14 @@ def test_step(self, *args, **kwargs):
     def predict(self, *args, **kwargs):
         return self.lightning_module.predict(*args, **kwargs)
 
-    def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: str) -> None:
+    def save_checkpoint(self, filepath: str, weights_only: bool = False) -> None:
```
Comment (Borda) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:206]: @tchaton @kaushikb11 where/when we changed this API?

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:206]: Changed in this PR: https://github.com/PyTorchLightning/pytorch-lightning/commit/646cf2f7d4876b5bf53b39960b2b82b722767ada
You can use git blame to find the associated PR.

**Review by tchaton — APPROVED** [review]
LGTM ! Great job.

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:206]: yes I know I just say that the API is different / not back compatible

**Pull request merged**